packages/rum-core/src/domain/rumSessionManager.ts

```diff
@@ -35,8 +35,15 @@ export const enum RumTrackingType {
 }
 
 export function startRumSessionManager(configuration: RumConfiguration, lifeCycle: LifeCycle): RumSessionManager {
-  const sessionManager = startSessionManager(configuration.cookieOptions, RUM_SESSION_KEY, (rawTrackingType) =>
-    computeSessionState(configuration, rawTrackingType)
+  if (!configuration.sessionStoreStrategyType) {
+    throw new Error('Cannot initialize RUM Session Manager without a storage.')
+  }
```

Comment:
At this stage - `configuration.sessionStoreStrategyType` is never undefined, so this throw should never trigger.
- Writing the configuration type so that `sessionStoreStrategyType` can only be defined is possible and allows to remove this check. However, it does not seem simple for the caller to match this type (unless exporting the type and casting with `as`). Related: https://github.com/microsoft/TypeScript/issues/31755
- We could add `sessionStoreStrategyType` to the function input parameters, but this looks redundant as we already have the info inside the configuration. If that is the preferred solution, maybe we should extract all relevant configuration parameters. But that would be even more verbose.

Reply:
FMU, there are two things:
- on the RUM side, we should not even allow an undefined strategy to be in the configuration, we could prevent that by making the type more specific in rum configuration
- on the logs side, we can allow an undefined strategy in the configuration because it is supported but some components only work with cookie or local storage strategy. For that, two options in mind, we could:
  - trust the caller (it is our own code 🙂) and use `configuration.sessionStoreStrategyType!` when needed
  - pass explicitelty `sessionStoreStrategyType` as parameter of those components with allowed values in the signature

wdyt?

Reply:
For now - let's use non-null assertion. We'll improve in a future iteration.